src/core/raster/qgsrasterlayer.h

```diff
@@ -142,6 +142,13 @@ class CORE_EXPORT QgsRasterLayer : public QgsMapLayer, public QgsAbstractProfile
        */
       bool skipCrsValidation = false;
 
+      //! Sets to TRUE to load the default raster attribute table for providers that support it
+      bool loadDefaultRat = true;
```

Comment:
Is there any benefit to **always** lazy loading the table?

Reply:
I don't think so, for the native (embedded) RATs managed by the provider it will be useful to automatically load the RAT at creation time, the RAT will also be used when running identify on the raster, the RAT entries for the identified value will be shown in the identify results, so that (foe example) the user will know that she have clicked on a "Vegetation" cell or on a "Buildings" cell. Anyway I will think about lazy loading when the time comes.

I still need to develop the RAT loading part, for now these flags are unused (I will remove them from this PR).

In any event, the RAT loading part will be controlled by client code in the application, the expected behavior will be:

1. automatic RAT loading when loading a new raster
  - if a native RAT (managed by the provider) exists, load it and create the default renderer from the RAT
  - if the raster is file-based and a DBF RAT with the same basename of the raster file esists, load it and create the renderer from the RAT

2. user initiated RAT loading (RAT import)
  - the user will be allowed to select an existing DBF RAT file and import it into the raster

3. user initiated RAT saving (RAT export)
  - the user will be allowed to export an existing raster RAT to a DBF RAT
  - the user will be allowed to export an existing raster RAT to a native RAT if the provider supports it

The RAT has an `isDirty` flag that will allow to prompt the user before loosing the changes.